### PR TITLE
ggml-cpu: Fix Flash Attention softcap handling for padded KV tiles on CPU

### PR DESCRIPTION
**Pull request by lilong555**
### Bug trigger conditions

This fixes a CPU Flash Attention issue when all of the following are true:

* `logit_softcap` is enabled;
* no attention mask is provided;
* the KV length is not a multiple of `GGML_FA_TILE_KV`.

The CPU implementation processes KV data in fixed-size tiles. For a partial KV tile, padded KQ entries must remain `-INFINITY` so that they receive zero weight in softmax.

### Root cause

Previously, KV padding was applied before softcap processing.

For padded entries, this effectively resulted in:

```text
-INFINITY -> tanh(-INFINITY) -> -logit_softcap
```

This converts the padding sentinel into a finite value inside the valid softcap logit range.

The padded entries can then contribute to the softmax normalization term, even though the corresponding padded V values do not contribute useful output values. This changes the attention result.

The issue is most visible just above the KV tile boundary. With `GGML_FA_TILE_KV = 64` and `KV = 65`, the second tile contains one valid KV entry and 63 padded entries.

When a mask is present, the issue can be hidden because the mask later restores the padded positions to `-INFINITY`.

### Fix

Apply softcap before writing the KV padding entries to `-INFINITY`.

The resulting order is:

```text
QK GEMM
scale
softcap
set padded KV logits to -INFINITY
mask
softmax
```

This guarantees that padded KV entries are still `-INFINITY` when they reach softmax.

The softcap computation for valid logits is unchanged. The fix does not add any buffers or additional processing passes, and keeps the existing full-tile vectorized `tanh` operation.

### Regression tests

Added unmasked Flash Attention regression cases around the CPU KV tile boundary:

* `KV = 63`, `64`, and `65`;
* `logit_softcap = 0` and `1`;
* `F32` and `F16` KV types;
* `nb = 65`, covering both a full Q tile and a Q tail.

The unmasked cases are important because a mask can overwrite the incorrect finite padding logits with `-INFINITY` and hide the bug.

Before the fix, these two cases consistently failed:

```text
KV=65, logit_softcap=1, type_KV=F32
KV=65, logit_softcap=1, type_KV=F16
```

Observed NMSE was approximately:

```text
0.0072 - 0.0075
```

compared with the allowed threshold:

```text
5e-4
```

After the fix:

```text
CPU targeted tests:  12/12 passed
CUDA targeted tests: 12/12 passed
```

CUDA code is unchanged. Running the same regression cases on CUDA provides a cross-check that the tests describe the intended Flash Attention semantics rather than CPU-specific output.

### Validation

Validation was performed locally on Windows at commit `b49650a`, with the local `HEAD` and remote `master` both at `b49650a`.

```text
CPU backend tests:   19811/19811 passed
CUDA backend tests:  16123/16123 passed

CPU targeted tests:  12/12 passed
CUDA targeted tests: 12/12 passed

git diff --check:     passed
```

These results cover the local Windows CPU and CUDA configurations used for development. Other supported platforms and backends are left to the GitHub CI matrix.

### Perplexity

Perplexity was also checked with the tested Llama 2 model and showed no change before and after the patch.

The tested model does not use attention softcap, so this is only a regression check for the non-softcap Flash Attention path. The targeted backend tests above are the primary validation for the affected softcap path.

### llama-bench

`llama-bench` results showed both positive and negative variation across runs, with overlapping ranges. No consistent directional performance regression was observed.

Direct measurements around the affected operator varied by approximately:

```text
-3.89% to +3.46%
```

The results are consistent with normal run-to-run measurement variation; this PR does not claim a performance improvement.

## Requirements

* I have read and agree with the [[contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md).
* AI usage disclosure: **YES** — AI was used to help analyze the bug, review the proposed fix and regression-test coverage, and draft/refine the PR description. The patch and the reported test results were reviewed and verified locally by me.

### CODE REVIEW

File: tests/test-backend-ops.cpp [hunk 1]
```diff
@@ -10702,6 +10702,15 @@ static std::vector<std::unique_ptr<test_case>> make_test_cases_eval() {
         }
     }
 
+    // Softcap with unmasked KV tails around the CPU tile boundary.
+    for (int kv : { 63, 64, 65 }) {
+        for (float logit_softcap : { 0.0f, 1.0f }) {
+            for (ggml_type type_KV : { GGML_TYPE_F32, GGML_TYPE_F16 }) {
+                test_cases.emplace_back(new test_flash_attn_ext(128, 128, 4, {1, 1}, kv, 65, false, false, 0, logit_softcap, GGML_PREC_F32, type_KV, type_KV));
```
[the remark this answers is not in the document]
Comment (lilong555) [reply tests/test-backend-ops.cpp:10709]: Fixed in 8fa3055 by applying softcap before padding in the Spacemit RVV path as well
